### PR TITLE
Add `typing.Optional` on missing annotations

### PR DESCRIPTION
**Pull request by Kludex**
There was a previous comment by @tomchristie about adding `typing.Optional` where is missing in a single PR.

I can't find the comment.

Note: I need to add `no-implicit-optional` on mypy configuration.

### CODE REVIEW

**Comment by odiseo0** [comment]
I have a question why use 
`
import typing
`
instead of
`
from typing import Optional, Sequence, Callable
`
Is there a code style guide for contributing in Starlette?

**Comment by Kludex** [comment]
> I have a question why use 
> `
> import typing
> `
> instead of
> `
> from typing import Optional, Sequence, Callable
> `
> Is there a code style guide for contributing in Starlette?

It's just to comply with the current pattern you see around in the code.

**Comment by odiseo0** [comment]
> > I have a question why use
> > `import typing`
> > instead of
> > `from typing import Optional, Sequence, Callable`
> > Is there a code style guide for contributing in Starlette?
> 
> It's just to comply with the current pattern you see around in the code.

Understood, thanks

**Review by adriangb — APPROVED** [review]

**Comment by Kludex** [comment]
I've added `no_implicit_optional`, so we don't have a future regression on this subject. I've also ignored this rule on `testclient.py`, as we are _thinking_ about changing the implementation there.

@adriangb Does your approval stands after my latest changes?

File: setup.cfg [hunk 1]
```diff
@@ -5,8 +5,12 @@ max-line-length = 88
 [mypy]
 disallow_untyped_defs = True
 ignore_missing_imports = True
+no_implicit_optional = True
 show_error_codes = True
 
+[mypy-starlette.testclient]
+no_implicit_optional = False
```
Comment (adriangb) [inline setup.cfg:12]: 👍 good call

**Review by adriangb — APPROVED** [review]
Yep I think this is good. I gave this a scroll through, I didn't see anything wrong so if MyPy is happy, I'm happy. LGTM!

**Pull request merged**